models/percona_sso_model_helpers_test.go

```diff
@@ -38,26 +40,62 @@ func setupDB(t *testing.T) (*reform.DB, func()) {
 }
 
 func TestPerconaSSODetails(t *testing.T) {
-	db, cleanup := setupDB(t)
-	defer cleanup()
-
-	expectedSSODetails := &models.PerconaSSODetails{}
-	err := models.InsertPerconaSSODetails(db.Querier, expectedSSODetails)
-	require.NoError(t, err)
-	ssoDetails, err := models.GetPerconaSSODetails(db.Querier)
-	require.NoError(t, err)
-	assert.NotNil(t, ssoDetails)
-	assert.Equal(t, expectedSSODetails.ClientID, ssoDetails.ClientID)
-	assert.Equal(t, expectedSSODetails.ClientSecret, ssoDetails.ClientSecret)
-	assert.Equal(t, expectedSSODetails.IssuerURL, ssoDetails.IssuerURL)
-	assert.Equal(t, expectedSSODetails.Scope, ssoDetails.Scope)
-	err = models.DeletePerconaSSODetails(db.Querier)
-	require.NoError(t, err)
-	ssoDetails, err = models.GetPerconaSSODetails(db.Querier)
-	assert.Error(t, err)
-	assert.Nil(t, ssoDetails)
-	// See https://github.com/percona/pmm-managed/pull/852#discussion_r738178192
-	ssoDetails, err = models.GetPerconaSSODetails(db.Querier)
-	assert.Error(t, err)
-	assert.Nil(t, ssoDetails)
+	ctx := context.Background()
+
+	t.Run("CorrectCredentials", func(t *testing.T) {
+		clientID, clientSecret := os.Getenv("OAUTH_PMM_CLIENT_ID"), os.Getenv("OAUTH_PMM_CLIENT_SECRET")
+		if clientID == "" || clientSecret == "" {
+			t.Skip("Environment variables OAUTH_PMM_CLIENT_ID / OAUTH_PMM_CLIENT_SECRET are not defined, skipping test")
+		}
+
+		db, cleanup := setupDB(t)
+		defer cleanup()
+
+		expectedSSODetails := &models.PerconaSSODetails{
+			IssuerURL:    "https://id-dev.percona.com/oauth2/aus15pi5rjdtfrcH51d7/v1/token?grant_type=client_credentials&scope=",
+			ClientID:     clientID,
+			ClientSecret: clientSecret,
+			Scope:        "percona",
+		}
+		InsertSSODetails := &models.PerconaSSODetailsInsert{
+			IssuerURL:    expectedSSODetails.IssuerURL,
+			ClientID:     expectedSSODetails.ClientID,
+			ClientSecret: expectedSSODetails.ClientSecret,
+			Scope:        expectedSSODetails.Scope,
+		}
+		err := models.InsertPerconaSSODetails(ctx, db.Querier, InsertSSODetails)
+		require.NoError(t, err)
+		ssoDetails, err := models.GetPerconaSSODetails(ctx, db.Querier)
+		require.NoError(t, err)
+		assert.NotNil(t, ssoDetails)
+		assert.Equal(t, expectedSSODetails.ClientID, ssoDetails.ClientID)
+		assert.Equal(t, expectedSSODetails.ClientSecret, ssoDetails.ClientSecret)
+		assert.Equal(t, expectedSSODetails.IssuerURL, ssoDetails.IssuerURL)
+		assert.Equal(t, expectedSSODetails.Scope, ssoDetails.Scope)
```

Comment:
Can we check that access_token has been fetched?

Reply:
Check added that access token and properties are not nil.